conans/model/ref.py

```diff
@@ -4,11 +4,23 @@
 from conans.model.version import Version
 
 
+# TODO: Move to some tools module
+# TODO: Use Conan way to skip coverage for py2/3 version
+try:  # pragma: no py3 cover
+    basestring  # attempt to evaluate basestring
+
+    def isstr(s):
+        return isinstance(s, basestring)
+except NameError:  # pragma: no py2 cover
+    def isstr(s):
+        return isinstance(s, (str, bytes))
+
+
 class ConanName(object):
     _max_chars = 50
     _min_chars = 2
     _validation_pattern = re.compile("^[a-zA-Z0-9_][a-zA-Z0-9_\+\.-]{%s,%s}$"
-                                     % (_min_chars - 1, _max_chars))
+                                     % (_min_chars - 1, _max_chars - 1))
```

Comment:
This is breaking! Should we change `_max_chars = 51` too?

Reply:
Could you add some test checking the length limit of the names (or add a new one checking the upper limit)?

Reply:
There are tests to check upper and lower limits here: https://github.com/conan-io/conan/pull/3410/files#diff-053aa4b3e517a282f2921d87da9db206R61, maybe in order to be backwards compatible we can make `_max_chars = 51` (or greater) and be sure that it is checked and used in strings with this value.

Reply:
I would say yes, we should raise it to 51